### PR TITLE
Battle 2k3: Add CBA single target ranged weapons

### PR DESCRIPTION
**Pull request by rueter37**
This PR adds CBA ranged weapons which target a single enemy and travel straight to it. The following checkboxes from #1269 are handled by it: "Use" checkbox, Dagger "fly in a straight line like a knife", Single Enemy and all Flying speed checkboxes.

### CODE REVIEW

**Review by fdelapena — APPROVED** [review]

File: src/scene_battle_rpg2k3.cpp [hunk 1]
```diff
@@ -1901,7 +1901,7 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 
 	if (source->GetType() == Game_Battler::Type_Ally) {
 		auto* sprite = static_cast<Game_Actor*>(source)->GetActorBattleSprite();
-		if (sprite && !sprite->IsIdling() && battle_action_state != BattleActionState_CBAMove && battle_action_state != BattleActionState_Animation) {
+		if (sprite && !sprite->IsIdling() && battle_action_state != BattleActionState_CBAMove && battle_action_state != BattleActionState_StartAnimation && battle_action_state != BattleActionState_CBARangedWeaponInit && battle_action_state != BattleActionState_CBARangedWeaponMove && battle_action_state != BattleActionState_Animation) {
```
Comment (Ghabry) [inline src/scene_battle_rpg2k3.cpp:1904]: This very long ``battle_action_state`` condition could be changed into a switch-case:

```
switch (battle_action_state) {
  case BattleActionState_CBAMove:
  case BattleActionState_StartAnimation:
   etc etc etc...
    return BattleActionReturn::eWait;
  default:;
```

**Comment by Ghabry** [comment]
I would assume that the "bow" option is just some kind of sinus curve? Or is there also a behaviour change?

**Comment by rueter37** [comment]
If you mean the boomerang path type with "bow" option, then unfortunately it is very difficult (see https://github.com/easyrpg/player/issues/1269#issuecomment-826091681).

File: src/scene_battle_rpg2k3.cpp (hunk 1 above)
Comment (rueter37) [reply src/scene_battle_rpg2k3.cpp]: Did the change now.

**Review by fmatthew5876 — APPROVED** [review]
Looks reasonable to me. Needs to be tested by someone other than code author.

**Review by Ghabry — COMMENTED** [review]
I found one minor problem:

When the attack count is set to x2 or x3 and the first enemy dies than the arrow is still targetting the dead enemy. in RPG_RT the arrow targets the next enemy.

**Comment by rueter37** [comment]
Fixed the targeting of the ranged weapon animation now. The weapon needs the "Attack All" flag set to make the ranged weapon animation target the original single target. Otherwise the current target is targeted by the ranged weapon animation. This applies to ranged weapons with the attack range set to "Single Enemy".

**Review by Ghabry — APPROVED** [review]
Works perfect now. Even better than in RPG_RT ;).

**Pull request merged**